lib/schema-controller.js

```diff
@@ -0,0 +1,123 @@
+'use strict'
+
+const { buildSchema } = require('./schemas')
+const {
+  ValidatorSelector,
+  SerializerCompiler: buildDefaultSerializer
+} = require('./schema-compilers')
+
+const defaultControllerOpts = {
+  bucket: buildSchema,
+  compilersFactory: {
+    buildValidator: undefined, // load once per fastify server start
+    buildSerializer: buildDefaultSerializer
+  }
+}
+
+module.exports = SchemaController
+module.exports.buildSchemaController = buildSchemaController
+module.exports.buildContext = buildContext
+
+/**
+ * Called at every fastify context that is being created.
+ * @param {object} parentSchemaCtrl: the SchemaController instance of the Fastify parent context
+ * @return {object}:a new SchemaController
+ */
+function buildSchemaController (opts) {
+  const option = opts || {
+    bucket: buildSchema,
+    compilersFactory: {
+      buildValidator: ValidatorSelector(),
+      buildSerializer: buildDefaultSerializer
+    }
+  }
+
+  return new SchemaController(option)
+}
+
+function buildContext (parentSchemaCtrl) {
+  const contextSchemaCtrl = new SchemaController(parentSchemaCtrl.opts)
+  contextSchemaCtrl.schemaBucket.store = parentSchemaCtrl.getSchemas() // TODO
+  contextSchemaCtrl.setValidatorCompiler(parentSchemaCtrl.getValidatorCompiler())
+  contextSchemaCtrl.setSerializerCompiler(parentSchemaCtrl.getSerializerCompiler())
+  contextSchemaCtrl.parent = parentSchemaCtrl
+  return contextSchemaCtrl
+}
+
+function SchemaController (option) {
```

Comment:
I think this is supposed to be called with `new`, right? Then let's make it a class.